File: docs/sphinx/targets/python/photonics.py
import cudaq

cudaq.set_target("photonics")


@cudaq.kernel
def photonicsKernel():
    qutrits = [qudit(3) for _ in range(2)]
    create(qutrits[0])
    create(qutrits[1])
    create(qutrits[1])
    mz(qutrits)


counts = cudaq.sample(photonicsKernel)
print(counts)

state = cudaq.get_state(photonicsKernel)
print(state)

File: docs/sphinx/targets/cpp/photonics.cpp
// Compile and run with:
// ```
// nvq++ --target orca-photonics photonics.cpp
// ./a.out
// ```

#include "cudaq/photonics.h"
#include "cudaq.h"

struct photonicsKernel {
  void operator()() __qpu__ {
    cudaq::qvector<3> qumodes(2);
    create(qumodes[0]);
    create(qumodes[1]);
    create(qumodes[1]);
    mz(qumodes);
  }
};

int main() {

  auto counts = cudaq::sample(photonicsKernel{});
  counts.dump();

  auto state = cudaq::get_state(photonicsKernel{});
  state.dump();
}